dash_auth/public_routes.py

```diff
@@ -0,0 +1,109 @@
+import inspect
+import os
+
+from dash import Dash, callback
+from dash._callback import GLOBAL_CALLBACK_MAP
+from dash import get_app
+from flask import Flask
+from werkzeug.datastructures import ImmutableDict
+from werkzeug.routing import Map, Rule
+
+# Add PUBLIC_ROUTES in the default Flask config
+DASH_PUBLIC_ASSETS_EXTENSIONS = "js,css"
+BASE_PUBLIC_ROUTES = [
+    f"/assets/<path:path>.{ext}"
+    for ext in os.getenv(
+        "DASH_PUBLIC_ASSETS_EXTENSIONS",
+        DASH_PUBLIC_ASSETS_EXTENSIONS,
+    ).split(",")
+] + [
+    "/_dash-component-suites/<path:path>",
+    "/_dash-layout",
+    "/_dash-dependencies",
+    "/_favicon.ico",
+    "/_reload-hash",
+]
+PUBLIC_ROUTES = "PUBLIC_ROUTES"
+PUBLIC_CALLBACKS = "PUBLIC_CALLBACKS"
+
+default_config = Flask.default_config
+Flask.default_config = ImmutableDict(
+    **default_config,
+    **{
+        PUBLIC_ROUTES: Map([]).bind(""),
+        PUBLIC_CALLBACKS: [],
```

Comment:
Do we need to touch `Flask.default_config`? Feels like `PUBLIC_ROUTES` could go on the app itself (`app._PUBLIC_ROUTES` perhaps?) and `PUBLIC_CALLBACKS` could just be a module-level variable here, and that would keep our impact a bit more contained. Would that work?

Reply:
Are you happy for me to set new attributes on the `Dash` instance like this?

Another option would be to not set default values and check their existence in the server config before using them. I think I'd go for that option if we want to have less impact.

Reply:
Excellent, I like that solution a lot!